lib/adagrams.rb

```diff
@@ -0,0 +1,199 @@
+require 'ap'
+require 'csv'
+require 'awesome_print'
+require 'pry'
+
+
+
+#wave 1
+
+def draw_letters()
+
+#static hash -- easy to change for future developer
+  static_letter_hash = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1
+  }
+
+#creates a letter_pool array from hash, to maintain letter/quantity odds, also to devloper to change hash orginal quanity value of letters.
+  letter_pool = []
+
+# iterating hash into letter pool array.
+  static_letter_hash.each do |letter, quantity|
+    quantity.times do
+      letter_pool << letter.to_s
+    end
+  end
+  player_hand = []
+
+
+#randomly pulls from letter pool and puts 10 cards in players hand
+
+  10.times do
+    random_index = rand(0..letter_pool.length)
+    player_hand << letter_pool[random_index]
+    letter_pool.slice!(random_index)
+  end
+  player_hand.each do |i|
+    letter_pool << i
+  end
+  return player_hand
+end
+
+#wave 2
+
+def uses_available_letters(input, letters_in_hand)
+
+#takes input word and converts into a array for each letter
+  input_split_array = input.scan(/\w/)
+
+#compares each letter in input word to see if it's in the player_hand array and deltes letter in hand to make sure it's not used to twice. return false if letter from input word is not in the player_hand.
+  input_split_array.each do |letter|
+    if letters_in_hand.include?(letter)
+      del_index = letters_in_hand.index(letter)
+      letters_in_hand.slice!(del_index)
+    else
+      return false
+    end
+  end
+  return true
+end
+
+
+# wave 3
+
+def score_word (word)
```

Comment:
just being nitpicky about syntax, but there technically shouldn't be a space between the method name and the opening parens for paramaters, aka: `score_word (word)` -> `score_word( word )`